yarn-project/prover-client/src/prover-pool/prover-pool.ts

```diff
@@ -0,0 +1,33 @@
+import { MemoryProvingQueue } from './memory-proving-queue.js';
+import { type ProvingAgent } from './prover-agent.js';
+import { type ProvingQueue } from './proving-queue.js';
+
+/**
+ * Utility class that spawns N prover agents all connected to the same queue
+ */
+export class ProverPool {
+  private agents: ProvingAgent[] = [];
+
+  constructor(
+    private size: number,
+    private agentFactory: (i: number) => ProvingAgent | Promise<ProvingAgent>,
+    public readonly queue: ProvingQueue = new MemoryProvingQueue(),
+  ) {}
+
+  async start(): Promise<void> {
+    this.agents = [];
```

Comment:
Shall we put some protection against people calling start for a second time without calling stop?

Reply:
Should I be able to start/stop this pool repeatedly? At the moment it's unclear. e.g. If I called start, stop, start, it would throw 'Prover pool is already running'. Which it's not.

Reply:
I have updated the code to track its state through a `isRunning` boolean.